### PR TITLE
replace mirage.js with msw.js mocking

### PR DESCRIPTION
**Pull request by kdembler**
The mocking code contains a couple of `@ts-ignore` because it was too much work to handle all the typescript cases. In general, the type safety should be much higher than that of previous code anyway

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # This file is commited. Do not store secrets here
 REACT_APP_ENV=staging
+#REACT_APP_ENV=development
```
Comment (drillprop) [inline .env:3]: should we uncomment this? Or at least update scripts `"dev": "cross-env REACT_APP_ENV=development yarn start"`.

**Review by drillprop — COMMENTED** [review]
Wow, that's a lot of code,  great work 🎉 Didn't find anything, in general looks good to me.

File: .env (hunk 1 above)
Comment (drillprop) [reply .env:3]: Also, I just noticed that we don't have `cross-env` anymore, so it should be installed for this

Comment (kdembler) [reply .env:3]: I'm not sure tbh. Do you think we should use `development` env by default?

Comment (drillprop) [reply .env:3]: Okay, we could just leave this as it is.

Comment (eldiegod) [reply .env:3]: I like the `"dev": "cross-env REACT_APP_ENV=development yarn start"` bit, commenting uncommenting would be more annoying imo

Comment (eldiegod) [reply .env:3]: also harder to figure out for new people looking at the repo

File: public/mockServiceWorker.js [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+}
+
+function respondWithMock(clientMessage) {
+  return new Response(clientMessage.payload.body, {
+    ...clientMessage.payload,
+    headers: clientMessage.payload.headers,
+  })
+}
+
+function uuidv4() {
```
Comment (eldiegod) [inline public/mockServiceWorker.js:304]: This could be pretty useful in the utils folder :)

File: src/mocking/accessors/pagination.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  after?: string | null
+}
+
+type OffsetLimitPaginatedDataArgs = {
+  offset?: number
+  limit?: number
+}
+
+const indexToCursor = (idx: number): string => btoa(idx.toString())
+const cursorToIndex = (cursor: string): number => +atob(cursor)
```
Comment (eldiegod) [inline src/mocking/accessors/pagination.ts:27]: Can u talk a bit about whats going on here? especially the `+atob(cursor)` part

File: src/mocking/data/mockVideos.ts [hunk 4]
```diff
@@ -1,20 +1,42 @@
...
+    category: mockCategories[idx % mockCategories.length],
+    license: mockLicenses[idx % mockLicenses.length],
+  } as MockVideo
 })
 
+const coverMockVideo: MockVideo = {
+  ...rawCoverVideo.video,
+  createdAt: new Date(rawCoverVideo.video.createdAt),
+  channel: coverMockChannel,
+  // @ts-ignore __typename not matching, whatever
```
Comment (eldiegod) [inline src/mocking/data/mockVideos.ts:32]: I know that feeling 😂

**Review by mikkio-j — COMMENTED** [review]
Wow, great work on that one 🔥 

It seems that the bug with not clickable videos thumbnails is back though.

**Comment by kdembler** [comment]
That's because the branch wasn't rebased, should be fine now

**Review by eldiegod — APPROVED** [review]
Looks good to me overall just a couple questions 👍

File: public/mockServiceWorker.js (hunk 2 above)
Comment (kdembler) [reply public/mockServiceWorker.js:304]: This is an autogenerated file so let's leave it as it is, if we need this functionality later we can copy it

File: src/mocking/accessors/pagination.ts (hunk 3 above)
Comment (kdembler) [reply src/mocking/accessors/pagination.ts:27]: Sure, `atob` is a function to decode a base64-encoded string. The `+` is an operator that allows you to convert string to number.
```js
const originalIndex = 3
const encodedIndexStr = btoa(originalIndex.toString()) // "Mw=="
const decodedIndexStr = atob(encodedIndexStr) // "3"
const index = +decodedIndexStr // 3
```

File: src/mocking/data/mockVideos.ts (hunk 4 above)
Comment (kdembler) [reply src/mocking/data/mockVideos.ts:32]: Haha yeah, it was just too much to handle everything, I think I already pushed my Typescript abilities to their limit anyway :P

File: src/mocking/accessors/pagination.ts (hunk 3 above)
Comment (eldiegod) [reply src/mocking/accessors/pagination.ts:27]: oh pretty cool

File: .env (hunk 1 above)
Comment (kdembler) [reply .env:3]: Let's leave it as it is for now

**Pull request merged**